src/main/old-save.gd

```diff
@@ -80,6 +80,90 @@ func transform_old_save() -> void:
+		PlayerSave.PLAYER_DATA_VERSION:
+			is_old = false
+		"15d2":
+			is_old = true
+		_:
+			push_warning("Unrecognized save data version: '%s'" % version_string)
+	return is_old
+
+
+"""
+Extracts a version string from the specified json save items.
+"""
+func get_version_string(json_save_items: Array) -> String:
+	var version: SaveItem
+	for json_save_item_obj in json_save_items:
+		var save_item: SaveItem = SaveItem.new()
+		save_item.from_json_dict(json_save_item_obj)
+		if save_item.type == "version":
+			version = save_item
+			break
+	return version.value if version else ""
+
+
+"""
+Transforms the specified json save items to the latest format.
+"""
+func transform_old_save_items(json_save_items: Array) -> Array:
+	var version_string := get_version_string(json_save_items)
+	if version_string == "15d2":
+		for json_save_item_obj in json_save_items:
+			var save_item: SaveItem = SaveItem.new()
+			save_item.from_json_dict(json_save_item_obj)
+			match save_item.type:
+				"scenario-history":
+					match save_item.key:
+						"rank-7k": attach_score_success(save_item, 200)
+						"rank-6k": attach_score_success(save_item, 200)
+						"rank-5k": attach_score_success(save_item, 750)
+						"rank-4k": attach_score_success(save_item, 300)
+						"rank-3k": attach_score_success(save_item, 999)
+						"rank-2k": attach_time_under_success(save_item, 180)
+						"rank-1k": attach_score_success(save_item, 1000)
+						"rank-1d": attach_score_success(save_item, 1200)
+						"rank-2d": attach_time_under_success(save_item, 300)
+						"rank-3d": attach_score_success(save_item, 750)
+						"rank-4d": attach_score_success(save_item, 2000)
+						"rank-5d": attach_score_success(save_item, 2500)
+						"rank-6d": attach_time_under_success(save_item, 180)
+						"rank-7d": attach_score_success(save_item, 4000)
+						"rank-8d": attach_score_success(save_item, 4000)
+						"rank-9d": attach_score_success(save_item, 6000)
+						"rank-10d": attach_score_success(save_item, 15000)
+				"version":
+					json_save_item_obj["value"] = PlayerSave.PLAYER_DATA_VERSION
+	return json_save_items
+
+
+"""
+Adds a 'success' key for a RankResult if the player met a target score and didn't lose
+"""
+func attach_score_success(save_item: SaveItem, target_score: int) -> void:
+	for rank_result_obj in save_item.value:
+		var rank_result_dict: Dictionary = rank_result_obj
+		rank_result_dict["success"] = not rank_result_dict.get("lost", true) \
+				and rank_result_dict.get("score", 0) >= target_score
+
+
+"""
+Adds a 'success' key for a RankResult if the player met a target time and didn't lose
+"""
+func attach_time_under_success(save_item: SaveItem, target_time: float) -> void:
+	for rank_result_obj in save_item.value:
+		var rank_result_dict: Dictionary = rank_result_obj
+		rank_result_dict["success"] = not rank_result_dict.get("lost", true) \
+				and rank_result_dict.get("seconds", 0.0) <= target_time
+
+
 """
 Appends a 'compare' flag to the ultra records.
```

Comment:
It might be good to flag these changes with the version they're modifying (15d2, 0517)

Reply:
Method names are now suffixed with `for_15d2`. The other 0517 method was already suffixed.